### DISCUSSION TITLE
Chamfer

### DISCUSSION THREAD

**Discussion by jgomez720** [KittyCAD/modeling-app#502] [New Feature Requests]
### Description
A tool that enables the user to chamfer an edge. This is similar to a fillet, except instead of a rounded edge you get a diagonal edge. Usually holds the same purpose, to make the part safer to hold. 

### Functionality
1. The user selects the _chamfer_ button.
2. The user selects the edges they want to chamfer
3. The user selects the dimensions they want to chamfer (this can be a dimension and an angle or two dimensions that relate the X and Y direction dimensions.
4. Confirm

OR 

1. The user selects the edges.
2. The user selects the _chamfer_ button.
3. The user inputs the dimensions. 
4. Confirm.

### Supporting Docs/Images
![Screenshot 2023-11-07 at 3 10 31 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/712b441c-51fb-4e30-836b-2c8758c84e62)
![Screenshot 2023-11-07 at 3 10 49 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/2d363d2e-e73f-4fb7-be5b-f2646e11dc11)
![Screenshot 2023-11-07 at 3 10 57 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/5eee00e1-eedb-43a1-80f3-bfb0ab9a2464)

## PseudoKCL
Very similar to `fillet` except one difference. For v1, we are going to assume all fillets are symmetric (same radius all the way through the fillet), however, for chamfer, we need to have a little more control. Asymmetric chamfers are created all the time. Here's an example of a symmetric one and asymmetric one.

**Symmetric**
```
const box = startSketchOn('XY')
  |> startProfileAt([0, 0], %)
  |> line([0, 5], tag: 'chamfer1', %)
  |> line([5, 0], tag: 'chamfer2', %)
  |> line([0, -5], tag: 'chamfer3', %)
  |> close( tag: 'chamfer4', %)
  |> extrude(5, %)

chamfer({
  symmetric: true, // not required, symmetric by default.
  length: 0.50, // only takes a single number if symmetric: true
  paths: ['chamfer1', 'chamfer2', 'chamfer3', 'chamfer4']
})
```
**Result:**
![Screenshot 2023-11-07 at 3 17 39 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/4499d0b9-a21d-477f-963c-82afd2f6dd7a)

**Asymmetric**
```
const box = startSketchOn('XY')
  |> startProfileAt([0, 0], %)
  |> line([0, 5], tag: 'chamfer1', %)
  |> line([5, 0], tag: 'chamfer2', %)
  |> line([0, -5], tag: 'chamfer3', %)
  |> close( tag: 'chamfer4', %)
  |> extrude(5, %)

chamfer({
  symmetric: false, 
  length: [0.5, 2.0], // must take two numbers if symmetric: false - horizontal dimension and vertical dimension
  paths: ['chamfer1', 'chamfer2', 'chamfer3', 'chamfer4']
})
```

**Result:**
![Screenshot 2023-11-07 at 3 18 22 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/b94a0c24-ff6d-4ca5-8b44-1134c16f0ba2)

**Comment by jessfraz** [comment]
covered by https://github.com/KittyCAD/modeling-app/pull/2681
